_posts/2020-08-13-convertextract-app.md

```diff
@@ -0,0 +1,95 @@
+
+# Who is involved with this project?
+
+* Kwak̓wala Corpus Collection group
+	- Sara Child [saratlilinukw@gmail.com](mailto:saratlilinukw@gmail.com)
+	- Daisy Rosenblum [daisy.rosenblum@ubc.ca](mailto:daisy.rosenblum@ubc.ca)
+	- Caroline Running Wolf [caroline.oldcoyote@gmail.com](mailto:caroline.oldcoyote@gmail.com)
+* App developer: [Aidan Pine](aidanpine.ca)
+* Support for adding mappings/parsers: [Fineen](fineen.davis@gmail.com)
+
+
+# What is needed to replicate the content in the post?
+
+- [g2p](https://github.com/roedoejet/g2p) mapping of the desired conversions
+- Language text to be converted
+- *Convertextract* application (read the post for installation!) 
+
+# What are the motivations behind this technology?
+
+As a Student Intern on the NRC's [Indigenous Language Technology (ILT) project](https://nrc.canada.ca/en/research-development/research-collaboration/programs/canadian-indigenous-languages-technology-project), I was approached by the Kwak̓wala Corpus Collection group to help create a systemic way to streamline the quality control process for their ELAN annotation data. Having many different people with many different orthographic conventions all working on annotating Kwak̓wala language data had resulted in inconsistencies.
+So, I added mappings in the [g2p library](https://github.com/roedoejet/g2p) 
+and a parser for ELAN in the Convertextract library. Aidan Pine then added a graphical user interface to Convertextract!
+
+# How to use the new Convertextract application for 'quality control' of ELAN annotations
+
+[Convertextract](https://github.com/roedoejet/convertextract), created by Aidan Pine, is a `python` library which extracts text data and finds/replaces specific text based on arbitrary correspondences. 
+Until now, only a CLI (Command Line Interface) was supported. Using Convertextract in the CLI allowed the user to convert a file based on pre-existing Mappings in the [g2p library](https://github.com/roedoejet/g2p) or based on a custom mapping. However, the downside is that some programming knowledge is needed to use the CLI.
+The latest update now includes a GUI (Graphical User Interface) in the form of an application (for Mac **only**). The app makes Convertextract more accessible for non-programmers.
+
+### 1. G2P mapping
+G2P mappings (i.e. arbitrary conversions between an input and an output) can be created manually as a `.csv` or `.json` file, or with the [g2p studio app](g2p-studio.herokuapp.com).
+Many languages are already supported. See the [g2p repository](https://github.com/roedoejet/g2p) on Github for detailed documentation on the G2P library.
+
+### 2. Language data
+
+Your language data must be in one of the supported file formats. The most recent addition is `.eaf` files, which allows ELAN annotations to be parsed and converted. For a full list of supported file types click [here](https://github.com/roedoejet/convertextract).
```

Comment:
```suggestion
Your language data must be in one of the supported file formats. The most recent addition is `.eaf` files, which allows ELAN annotations to be parsed and converted. For a full list of supported file types click [here](https://github.com/roedoejet/convertextract#converting-a-file-based-on-pre-existing-mappings-in-the-g2p-library).
```

So I followed the link, but I genuinely struggled to find a list of supported file formats amid all the noise. That list also omits `.eaf`. Is there a better list of the supported file formats?

Here's where I got the list, for reference:

> For this type of call, convertextract requires three arguments:
>
> 1. A file containing text to convert (as of Version 1.0.4, this includes **.pptx**, **.docx**, **.xlsx**, and **.txt**)